### ISSUE TITLE
support both 32-bit and 64-bit executables in Linux package

### ISSUE THREAD

**Issue by mykmelez** [mozilla/r2d2b2g#30]
[Bug 804330](https://bugzilla.mozilla.org/show_bug.cgi?id=804330) will add 64-bit Linux builds of B2G Desktop, and we should bundle the 64-bit build with the 32-bit build in the Linux package, selecting the executable to run at runtime based on the bit-depth of the user's Linux installation.

(We need to bundle the builds into the same package and select the executable to run at runtime because AMO, on which we plan to distribute the package, only supports a single Linux XPI.)

**Comment by fwenzel** [comment]
Will this make the download even more massive than it already is?

**Comment by tofumatt** [comment]
I'm assuming yes :-(

But there's no way around that unless we have a version of the add-on for each platform. With this many builds I'm starting to think that isn't a horrible idea…
- tofumatt | http://tofumatt.com

**Comment by mykmelez** [comment]
Yes, on Linux it'll increase the package size by another 20MB or so.

**Comment by mykmelez** [comment]
We do have separate packages for Windows, Mac, and Linux. And we can roll separate packages for 32-bit and 64-bit Linux. But we can't distribute them on AMO, because AMO doesn't support separate packages for 32-bit and 64-bit Linux.

**Comment by tofumatt** [comment]
Oh, well that's not so bad. Cool beans.

**Comment by mykmelez** [comment]
(AMO does support separate packages for Windows, Mac, and Linux, though, and it picks the package to provide the user automatically based on the user's OS, so this change will only affect Linux users. Windows and Mac users will continue to get Windows- and Mac-specific packages.)

**Comment by tofumatt** [comment]
Good to know! Given that I've only ever worked on purely cross-platform JS add ons, I wasn't aware of how AMO dealt with OS-specific stuff.

Thanks.

**Comment by mykmelez** [comment]
Fixed in f9d03a0ab70697d2c5eb4f7422c7d184787d05eb, but note that you need to run an extra command to build a package that contains both the 32-bit and 64-bit executables. On a 32-bit Linux system, do:

```
make
python ./build.py --platform linux64
make package
```

On a 64-bit system, do:

```
make
python ./build.py --platform linux
make package
```

In each case, the call to `make` retrieves the binary for the architecture on which you are running the command, while the call to `python ./build.py` retrieves it for the other architecture.
